tests/e2e/creation_test.go

```diff
@@ -351,3 +358,112 @@ func (tc *DSCTestCtx) UpdateRegistriesNamespace(targetNamespace, expectedValue s
 		WithCustomErrorMsg("Failed to update RegistriesNamespace to %s, expected %s", targetNamespace, expectedValue),
 	)
 }
+
+// ValidateComponentsDeploymentFailure simulates component deployment failure using restrictive resource quota.
+func (tc *DSCTestCtx) ValidateComponentsDeploymentFailure(t *testing.T) {
+	t.Helper()
+
+	var allComponents = []string{
+		"dashboard", "workbenches", "modelregistry", "ray", "kueue",
+		"codeflare", "trainingoperator", "datasciencepipelines",
+		"kserve", "modelmeshserving", "trustyai", "feastoperator", "llamastackoperator",
+	}
+
+	originalDSC := tc.FetchDataScienceCluster()
+	restrictiveQuota := createRestrictiveQuotaForOperator(tc.AppsNamespace)
+
+	t.Log("Creating restrictive resource quota in operator namespace")
+	tc.EnsureResourceCreatedOrPatched(WithObjectToCreate(restrictiveQuota))
+
+	t.Log("Enabling all components in DataScienceCluster")
+	tc.EnsureResourceCreatedOrPatched(
+		WithMinimalObject(gvk.DataScienceCluster, tc.DataScienceClusterNamespacedName),
+		WithMutateFunc(updateAllComponentsTransform(allComponents, operatorv1.Managed)),
+	)
+
+	t.Log("Verifying DSC reports all failed components due to restrictive quota")
+	tc.EnsureResourceExists(
+		WithMinimalObject(gvk.DataScienceCluster, tc.DataScienceClusterNamespacedName),
+		WithCondition(jqAllComponentsFailureMatcher(allComponents)),
+		WithCustomErrorMsg("DSC should report all components failed due to insufficient quota"),
+		WithEventuallyTimeout(tc.TestTimeouts.mediumEventuallyTimeout),
+	)
+
+	t.Log("Cleaning up restrictive quota")
+	tc.DeleteResource(WithObjectToCreate(restrictiveQuota))
+
+	t.Log("Restoring original DSC state")
+	tc.restoreDSC(t, originalDSC)
+
+	t.Log("Verifying DSC reports components ready after restoring")
+	tc.verifyComponentsReady(t)
```

Comment:
You should then disable all components otherwise they will be already enabled making the subsequent components tests misbehaving/useless.

Reply:
restoreDSC(t, originalDSC) is nothing but disabling all components. The original state of DSC is with all components disabled. i think this log message is misleading, i will correct this.